### ISSUE TITLE
Backward docking stucks on Humble

### ISSUE THREAD

**Issue by Jakubach** [open-navigation/opennav_docking#68]
Hello,  when using Backward docking mode the robot correctly reverse to the docking pose but very near docking pose it starts doing in-place circles without the end. The problem only occurs it this mode.
The solution I've found:

The `SmoothControlLaw` method used in docking server is `calculateRegularVelocity(pose, backward) `([link](https://github.com/grupo-avispa/navigation2/blob/940322af527f8284a218b53e63da8337e853f636/nav2_docking/opennav_docking/src/controller.cpp#L64C23-L64C47)) and changing it to that one taking also the robot current pose fixes the problem: `calculateRegularVelocity(const geometry_msgs::msg::Pose & target,const geometry_msgs::msg::Pose & current,const bool & backward = false) `([link](https://github.com/grupo-avispa/navigation2/blob/44a69ccde93514ed1a962b7f18cb101ec671d93a/nav2_graceful_controller/include/nav2_graceful_controller/smooth_control_law.hpp#L92C29-L92C53))
I haven't checked that but the problem can also occur in the `GracefulController` since it's implemented similarly ([link](https://github.com/ros-navigation/navigation2/blob/b71caaf46d8a95638863d8fb1d15b954894f8f54/nav2_graceful_controller/src/graceful_controller.cpp#L185)).
I also had to delete the lines with `backward_projection` (428-432) ([link](https://github.com/grupo-avispa/navigation2/blob/940322af527f8284a218b53e63da8337e853f636/nav2_docking/opennav_docking/src/docking_server.cpp#L428)) (then I modified it for my case to perform initial in-place rotations).

In summary:
- Used `calculateRegularVelocity(pose,current, backward)` instead of `calculateRegularVelocity(pose, backward)`
- Commented  out the part with `backward_projection`(and that part does not exist in `GracefulController`)

After this changes the robot stops correctly in the end point (with default controller parameters)

[Screencast from 11-07-2024 06_53_45 PM (1).webm](https://github.com/user-attachments/assets/5a766d09-83f1-42c7-b048-1a115fd0026c)

https://github.com/user-attachments/assets/793f4ab2-50b6-4a94-a25a-3e03f61c7740

I will propose that changes on pull request.

**Comment by ajtudela** [comment]
Hi, I think I have a clearly idea of what's happening to you:

- First, make sure that the dock orientation is pointing at the wall (not at the robot):

![image](https://github.com/open-navigation/opennav_docking/assets/4681902/8b60d172-1299-4033-b8c1-8c504c8753c8)

- Second, asumming that you haver sensors in the back to detect the dock, set the staging pose te reversed using the  **staging_yaw_offset**. 



*NOTES FOR ME*: Improve the documentation for this cases and implement detection+rotate+reverse ASAP.

**Comment by Jakubach** [comment]
Hi, thank you for your response. I've tried re-orientating the dock pose (rotation by PI around Z) so if the robot pose is [0,0 0] then dock pose is [0,0, PI]. 
To the second point, my robot has a sensor to detect the dock in a front. I was also playing with **staging_yaw_offset** but the dock detection starts after reaching the staging pose not during reaching that pose (then it could work).

**Comment by ajtudela** [comment]
Your case: sensors in the front and docking pads in the back is not currently supported. There is an open issue in nav2 [here](https://github.com/ros-navigation/navigation2/issues/4403).

I saw your PR. If you want to implement this feature, you need to do it in the main branch of Nav2 and I'll review it.

**Comment by Jakubach** [comment]
I made a PR in the main branch of Nav2

**Comment by SteveMacenski** [comment]
This all sounds good, thanks @ajtudela for responding to @Jakubach and helping to review! Its very appreciated :-)

**Comment by SteveMacenski** [comment]
Merge imminent in Nav2's updated version

**Comment by SteveMacenski** [comment]
This has been resolved in Nav2

**Comment by loudboy10** [comment]
What about an option to back away from a marker, reversing into the dock under guidance from sensors in the front? A forklift with a load maneuvering into a loading bay would be such a scenario. Navigate to a staging point perpendicular to the marker, rotate 90deg to face the marker, then back away from the marker to reach the dock.
